tests/integration.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(test)]
+mod tests {
+    use anyhow::Result;
+    use image_rs::image::ImageClient;
+    use std::path::Path;
+    use std::process::{Child, Command};
+
+    fn start_attestation_agent() -> Result<Child> {
+        let script_dir = format!("{}/{}", std::env!("CARGO_MANIFEST_DIR"), "scripts");
+        let aa_path = format!("{}/{}", script_dir, "attestation-agent");
+
+        if !Path::new(&aa_path).exists() {
+            let script_path = format!("{}/{}", script_dir, "build_attestation_agent.sh");
+            Command::new(script_path)
+                .output()
+                .expect("Failed to build attestation-agent");
+        }
+
+        Ok(Command::new(aa_path)
+            .args(&["--keyprovider_sock"])
+            .args(&["127.0.0.1:48888"])
+            .spawn()?)
+    }
+
+    #[tokio::test]
+    async fn test_image_rs() {
+        let mut aa = start_attestation_agent().expect("Failed to start attestation agent!");
+
+        let manifest_dir = std::env!("CARGO_MANIFEST_DIR");
+        let image = "docker.io/arronwang/busybox_kbs_encrypted";
```

Comment:
To avoid using your own docker hub user, how about creating a https://hub.docker.com/u/confidential-containers or similar ([like we have for Kata]( https://hub.docker.com/u/katadocker))?

Reply:
Thanks, good suggestion, I'll try to create an confidential-containers repo in docker hub.

Reply:
Ack - the name might be a problem (as admittedly `confidentialcontainers` is pretty hard to read) but it might also be safer long term to have a CC specific account.

Reply:
Hi @arronwy @jodh-intel , I would even avoid docker.io as some developers have complaint about hitting the limits to pull images. So I would rather use quay.io. In fact we already have a repository for CC images: https://quay.io/repository/kata-containers/confidential-containers

Reply:
cool, thanks!

Reply:
seems [quay.io](http://quay.io/) don't support encrypted container image format：
``` 
Error uploading manifest kbs_encrypted_unsigned to [quay.io/kata-containers/confidential-containers](http://quay.io/kata-containers/confidential-containers): manifest invalid: manifest invalid
Unknown media type during manifest conversion: "application/vnd.oci.image.layer.v1.tar+gzip+encrypted
```

Reply:
hi @arronwy, hmm... I will ask people about that. However I don't want to block his PR; so it's okay to me use docker.io temporarily